src/components/LibraryContainer.tsx

```diff
@@ -22,18 +25,37 @@ export class LibraryContainer extends React.Component<LibraryContainerProps, Lib
     constructor(props: LibraryContainerProps) {
 
         super(props);
+        this.state = { inSearchMode: false };
         this.generatedLibraryItems = buildLibraryItemsFromLayoutSpecs(
             this.props.loadedTypesJson, this.props.layoutSpecsJson);
     }
 
+    onSearchModeChanged(inSearchMode: boolean) {
+        this.setState({ inSearchMode: inSearchMode });
+    }
+
     render() {
 
         try {
             let index = 0;
```

Comment:
You can do this later, but let's move `index` into the `else` clause below before the `map`.